File: exp-zfrontend/src/api/bidding/bid.ts
import request from '@/api/request';
import {buildPageQuery, type PageQueryInput} from '@/api/common';

export type BidStatus =
  | 'PREPARE'
  | 'SUBMITTED'
  | 'EVALUATING'
  | 'WON'
  | 'LOST'
  | 'ABANDONED';

export interface BidListVO {
  bidId: string;
  tenderId?: string;
  tenderName?: string;
  supplierId?: string;
  supplierName?: string;
  bidCode?: string;
  bidName?: string;
  bidTotalAmount?: number;
  currency?: string;
  bidSubmitTime?: string;
  bidStatus: BidStatus | string;
  winFlag?: number;
  winNoticeNo?: string;
  projectId?: string;
  projectName?: string;
  createdBy?: string;
  createdByName?: string;
  managerPersonId?: string;
  managerPersonName?: string;
  orgId?: string;
  orgIdName?: string;
  salesmanId?: string;
  salesmanName?: string;
  createdTime?: string;
}

export interface BidDetailVO extends BidListVO {
  contractId?: string;
  createdDeptId?: string;
  createdDeptName?: string;
  createdPostId?: string;
  createdPostName?: string;
  updatedTime?: string;
  remark?: string;
}

export interface PageResult<T> {
  list: T[];
  total: number;
  page: number;
  size: number;
}

export interface QueryBidParams {
  purchaserName?: string;
  tenderName?: string;
  projectName?: string;
  bidCode?: string;
  bidName?: string;
  bidStatus?: BidStatus | string;
  winFlag?: number;
  bidSubmitTimeStart?: string;
  bidSubmitTimeEnd?: string;
  createdBy?: number;
  createdTimeStart?: string;
  createdTimeEnd?: string;
  sort?: string;
}

export interface CreateBidReq {
  tenderId: number;
  supplierId: number;
  bidCode: string;
  bidName: string;
  bidTotalAmount: number;
  currency: string;
  principalId?: number;
  salesmanId?: number;
  orgId: number;
  bidSubmitTime?: string;
  projectId?: number;
  remark?: string;
}

export interface UpdateBidReq {
  bidId: number;
  bidCode: string;
  bidName: string;
  bidTotalAmount: number;
  principalId?: number;
  salesmanId?: number;
  orgId: number;
  remark?: string;
}

export interface DeleteBidReq {
  bidId: number;
}

export interface BidStatusReq {
  bidId: number;
  bidStatus: BidStatus | string;
}

export interface CheckBidCodeParams {
  bidCode: string;
  excludeBidId?: number;
}

export interface CheckSupplierBidParams {
  tenderId: number;
  supplierId: number;
  excludeBidId?: number;
}

export function queryBidList(data: PageQueryInput<QueryBidParams>) {
  return request.post<PageResult<BidListVO>, PageResult<BidListVO>>('/exp/bid/bidding/list',
     buildPageQuery<QueryBidParams>(data),
  );
}

export function getBidDetail(bidId: number | string) {
  return request.get<BidDetailVO, BidDetailVO>('/exp/bid/bidding/detail', {
    params: { bidId: Number(bidId) },
  });
}

export function createBid(data: CreateBidReq) {
  return request.post<BidDetailVO, BidDetailVO>('/exp/bid/bidding/create', data);
}

export function updateBid(data: UpdateBidReq) {
  return request.post<BidDetailVO, BidDetailVO>('/exp/bid/bidding/update', data);
}

export function deleteBid(data: DeleteBidReq) {
  return request.post<void, void>('/exp/bid/bidding/delete', data);
}

export function updateBidStatus(data: BidStatusReq) {
  return request.post<BidDetailVO, BidDetailVO>('/exp/bid/bidding/status', data);
}

export function checkBidCode(params: CheckBidCodeParams) {
  return request.get<boolean, boolean>('/exp/bid/bidding/checkBidCode', { params });
}

export function checkSupplierBid(params: CheckSupplierBidParams) {
  return request.get<boolean, boolean>('/exp/bid/bidding/checkSupplierBid', { params });
}

File: exp-bid-contract/exp-bid-contract-core/src/main/java/jh/exp/bid/contract/core/entity/Bid.java
package jh.exp.bid.contract.core.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 投标主表，对应 exp_bid
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "exp_bid")
@TableName("exp_bid")
public class Bid {
    // 主键ID
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @TableId(type = IdType.AUTO)
    @Column(name = "bid_id")
    private Long bidId;

    // 招标项目ID
    @Column(name = "tender_id")
    private Long tenderId;

    // 投标单位ID，关联企业外部基础信息（合作单位/供应商）
    @Column(name = "supplier_id")
    private Long supplierId;

    // 投标编号（公司内部编码）
    @Column(name = "bid_code", length = 100)
    private String bidCode;

    // 投标名称
    @Column(name = "bid_name", length = 200)
    private String bidName;

    // 投标总报价金额
    @Column(name = "bid_total_amount", precision = 20, scale = 2)
    private BigDecimal bidTotalAmount;

    // 币种
    @Column(name = "currency", length = 10)
    private String currency;

    // 投标提交时间
    @Column(name = "bid_submit_time")
    private LocalDateTime bidSubmitTime;

    // 投标状态（准备、已提交、评审中、中标、未中标、放弃等）
    @Column(name = "bid_status", length = 32)
    private String bidStatus;

    // 是否中标标识（0/1）
    @Column(name = "win_flag")
    private Integer winFlag;

    // 中标通知书编号
    @Column(name = "win_notice_no", length = 100)
    private String winNoticeNo;

    // 合同ID，关联合同管理模块合同表
    @Column(name = "contract_id")
    private Long contractId;

    // 工程项目ID，关联工程项目服务模块
    @Column(name = "project_id")
    private Long projectId;

    // 创建人用户ID，关联账号信息表
    @Column(name = "created_by")
    private Long createdBy;

    // 创建人部门ID，关联部门管理
    @Column(name = "created_dept_id")
    private Long createdDeptId;

    // 创建人岗位ID，关联岗位管理
    @Column(name = "created_post_id")
    private Long createdPostId;

    // 创建时间
    @Column(name = "created_time")
    private LocalDateTime createdTime;

    // 更新时间
    @Column(name = "updated_time")
    private LocalDateTime updatedTime;

    // 备注
    @Column(name = "remark", length = 500)
    private String remark;
}
